File: backend/app/schemas/catalog.py
from __future__ import annotations

from pydantic import BaseModel, ConfigDict

from app.schemas.product import CategoryResponse, VendorResponse


class CategoryListResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    items: list[CategoryResponse]


class VendorListResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    items: list[VendorResponse]


class VendorStorefrontResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: str
    name: str
    slug: str
    is_active: bool
    is_verified: bool
    business_name: str | None
    activity_type: str | None
    storefront_tier: str | None
    city: str | None
    phone: str | None
    address: str | None
    description: str | None
    logo_url: str | None
    cover_image_url: str | None
    badge_label: str | None
    starting_price: float | None
    price_suffix: str | None
    highlight_items: list[str]
    product_count: int
    promotion_count: int
    service_count: int
    room_type_count: int


class VendorStorefrontListResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    items: list[VendorStorefrontResponse]


class PromotionItemResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    product_id: str
    product_name: str
    brand: str
    main_image_url: str | None
    category_slug: str | None
    vendor: VendorResponse
    original_amount: float
    promo_amount: float
    currency: str
    promo_until: str | None
    is_boosted: bool


class PromotionListResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    items: list[PromotionItemResponse]

File: frontend/src/types/catalog.ts
export type VendorStorefront = {
  id: string;
  name: string;
  slug: string;
  is_active: boolean;
  is_verified: boolean;
  business_name: string | null;
  activity_type: string | null;
  storefront_tier: string | null;
  city: string | null;
  phone: string | null;
  address: string | null;
  description: string | null;
  logo_url: string | null;
  cover_image_url: string | null;
  product_count: number;
  promotion_count: number;
  service_count: number;
  room_type_count: number;
};

export type CatalogCategory = {
  id: string;
  name: string;
  slug: string;
};

export type PromotionItem = {
  product_id: string;
  product_name: string;
  brand: string;
  main_image_url: string | null;
  category_slug: string | null;
  vendor: {
    id: string;
    name: string;
    slug: string;
    is_active: boolean;
    is_verified: boolean;
  };
  original_amount: number;
  promo_amount: number;
  currency: string;
  promo_until: string | null;
  is_boosted: boolean;
};
